src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service;
+
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Map<String, Integer> fruits = new HashMap<>();
```

Comment:
Consider using a thread-safe collection like `ConcurrentHashMap` if this service is intended to be used in a multi-threaded environment.